packages/manifold/contracts/gachaclaims/ERC1155GachaLazyClaim.sol

```diff
@@ -0,0 +1,250 @@
+  }
+
+  constructor(address initialOwner) GachaLazyClaim(initialOwner) {}
+
+  /**
+   * See {IERC1155GachaLazyClaim-initializeClaim}.
+   */
+  function initializeClaim(
+    address creatorContractAddress,
+    uint256 instanceId,
+    ClaimParameters calldata claimParameters
+  ) external payable override adminRequired {
+    if (instanceId == 0 || instanceId > MAX_UINT_56) revert IGachaLazyClaim.InvalidInstance();
+
+    // Checks
+    if (claimParameters.storageProtocol == StorageProtocol.INVALID) revert IGachaLazyClaim.InvalidStorageProtocol();
+    if (claimParameters.endDate != 0 && claimParameters.startDate >= claimParameters.endDate) revert IGachaLazyClaim.InvalidDate();
+    
+    address[] memory receivers = new address[](1);
+    receivers[0] = msg.sender;
+    uint256[] memory amounts = new uint256[](claimParameters.itemVariations);
+    string[] memory uris = new string[](claimParameters.itemVariations);
+    uint256[] memory newTokenIds = IERC1155CreatorCore(creatorContractAddress).mintExtensionNew(receivers, amounts, uris);
+
+    require(newTokenIds[0] <= MAX_UINT_80, "Token ID exceeds uint80 range");
+
+    // Create the claim
+    _claims[creatorContractAddress][instanceId] = Claim({
+      storageProtocol: claimParameters.storageProtocol,
+      total: 0,
+      totalMax: claimParameters.totalMax,
+      startDate: claimParameters.startDate,
+      endDate: claimParameters.endDate,
+      startingTokenId: uint80(newTokenIds[0]),
+      itemVariations: claimParameters.itemVariations,
+      location: claimParameters.location,
+      paymentReceiver: claimParameters.paymentReceiver,
+      cost: claimParameters.cost,
+      erc20: claimParameters.erc20
+    });
+    for (uint8 i; i < claimParameters.itemVariations; i++) {
+      _tokenInstances[creatorContractAddress][newTokenIds[i]] = instanceId;
+    }
+
+    emit GachaClaimInitialized(creatorContractAddress, instanceId, msg.sender);
+  }
+
+  /**
+   * See {IERC1155GachaLazyClaim-updateClaim}.
+   */
+  function updateClaim(
+    address creatorContractAddress,
+    uint256 instanceId,
+    ClaimParameters memory claimParameters
+  ) external override adminRequired {
+    Claim memory claim = _getClaim(creatorContractAddress, instanceId);
+    if (claimParameters.endDate != 0 && claimParameters.startDate >= claimParameters.endDate)
+      revert IGachaLazyClaim.InvalidDate();
+    if (claimParameters.totalMax != claim.totalMax) revert IGachaLazyClaim.CannotChangeTotalMax();
+    if (claimParameters.startingTokenId != claim.startingTokenId) revert IGachaLazyClaim.CannotChangeStartingTokenId();
+    if (claimParameters.itemVariations != claim.itemVariations) revert IGachaLazyClaim.CannotChangeItemVariations();
+    if (claimParameters.storageProtocol != claim.storageProtocol) revert IGachaLazyClaim.CannotChangeStorageProtocol();
+    if (keccak256(abi.encodePacked(claimParameters.location)) != keccak256(abi.encodePacked(claim.location))) {
```

Comment:
Why not?

Reply:
We need to in case there's an asset mistake.

Reply:
okay updated so everything but paymentToken (consistent with other extensions) and tokenVariations (f.k.a. itemVariations) can be updated

reasoning that tokenVariations would require funky logic and more indexing to handle if other tokens were minted in between gacha claim initialized and updated